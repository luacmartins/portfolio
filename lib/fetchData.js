import fs from 'fs'
import path from 'path'
import matter from 'gray-matter'

const projectsDirectory = path.join(process.cwd(), '/pages/projects')
const bioDirectory = path.join(process.cwd(), '/pages/bio')

export function getSortedProjectsData() {
   const fileNames = fs.readdirSync(projectsDirectory)
   const allProjectsData = fileNames.map(fileName => {
      // Remove ".md" from file name to get id
      const id = fileName.replace(/\.md$/, '')

      // Read markdown file as string
      const fullPath = path.join(projectsDirectory, fileName)
      const fileContents = fs.readFileSync(fullPath, 'utf8')

      // Use gray-matter to parse the post metadata section
      const matterResult = matter(fileContents)

      // Combine the data with the id
      return {
         id,
         ...matterResult.data
      }
   })
   // Sort posts by date
   return allProjectsData.sort((a, b) => {
      if (a.date < b.date) {
         return 1
      } else {
         return -1
      }
   })
}

export function getBioData() {
   const fileName = 'bio.md'
   const fullPath = path.join(bioDirectory, fileName)
   const fileContent = fs.readFileSync(fullPath, 'utf8')
   const matterResult = matter(fileContent)
   console.log(matterResult)
   return {
      id: 'bio',
      ...matterResult.data
   }
}