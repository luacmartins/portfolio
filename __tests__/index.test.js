import IndexPage from '../pages/index'
import renderer from 'react-test-renderer'
import { getSortedData } from '../lib/fetchData'

const allProjectsData = getSortedData('projects')
const bioData = getSortedData('bio')[0]
const allExperienceData = getSortedData('experience')
const allEducationData = getSortedData('education')

test('should render index page correctly', () => {
   const component = renderer.create(<IndexPage bioData={bioData} allProjectsData={allProjectsData} allExperienceData={allExperienceData} allEducationData={allEducationData} />)
   expect(component).toMatchSnapshot()
}) 