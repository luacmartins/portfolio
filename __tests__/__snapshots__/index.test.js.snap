// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render index page correctly 1`] = `
<div
  className="min-h-screen py-16 px-4 text-gray-800 lg:px-0 lg:py-0"
>
  <div
    className="flex items-center justify-center sm:px-24 lg:justify-start lg:pt-16 text-center flex-col lg:fixed lg:px-16 lg:w-104"
  >
    <img
      alt="profile picture"
      className="rounded-full w-48 shadow-lg"
      src="/images/profile.jpg"
    />
    <h2
      className="mt-8 text-2xl leading-tight font-thin"
    >
      Hi, my name is Carlos. I'm a Software Engineer.
    </h2>
    <div
      className="mt-4"
    >
      <div
        className="flex"
      >
        <a
          alt="github icon"
          className="h-6 w-6 mx-2 hover:opacity-75"
          href="https://github.com/luacmartins"
          target="_blank"
        >
          <svg
            className="fill-current h-full w-full"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m12 .5c-6.63 0-12 5.28-12 11.792 0 5.211 3.438 9.63 8.205 11.188.6.111.82-.254.82-.567 0-.28-.01-1.022-.015-2.005-3.338.711-4.042-1.582-4.042-1.582-.546-1.361-1.335-1.725-1.335-1.725-1.087-.731.084-.716.084-.716 1.205.082 1.838 1.215 1.838 1.215 1.07 1.803 2.809 1.282 3.495.981.108-.763.417-1.282.76-1.577-2.665-.295-5.466-1.309-5.466-5.827 0-1.287.465-2.339 1.235-3.164-.135-.298-.54-1.497.105-3.121 0 0 1.005-.316 3.3 1.209.96-.262 1.98-.392 3-.398 1.02.006 2.04.136 3 .398 2.28-1.525 3.285-1.209 3.285-1.209.645 1.624.24 2.823.12 3.121.765.825 1.23 1.877 1.23 3.164 0 4.53-2.805 5.527-5.475 5.817.42.354.81 1.077.81 2.182 0 1.578-.015 2.846-.015 3.229 0 .309.21.678.825.56 4.801-1.548 8.236-5.97 8.236-11.173 0-6.512-5.373-11.792-12-11.792z"
            />
          </svg>
        </a>
        <a
          alt="linkedin icon"
          className="h-6 w-6 mx-2 hover:opacity-75"
          href="https://www.linkedin.com/in/luacmartins/"
          target="_blank"
        >
          <svg
            className="fill-current h-full w-full"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M475.074 0H36.926C16.53 0 0 16.531 0 36.926v438.148C0 495.47 16.531 512 36.926 512h438.148C495.47 512 512 495.469 512 475.074V36.926C512 16.53 495.469 0 475.074 0zM181.61 387h-62.347V199.426h62.347zm-31.172-213.188h-.406c-20.922 0-34.453-14.402-34.453-32.402 0-18.406 13.945-32.41 35.274-32.41 21.328 0 34.453 14.004 34.859 32.41 0 18-13.531 32.403-35.274 32.403zM406.423 387h-62.34V286.652c0-25.218-9.027-42.418-31.586-42.418-17.223 0-27.48 11.602-31.988 22.801-1.649 4.008-2.051 9.61-2.051 15.215V387h-62.344s.817-169.977 0-187.574h62.344v26.558c8.285-12.78 23.11-30.96 56.188-30.96 41.02 0 71.777 26.808 71.777 84.421zm0 0"
            />
          </svg>
        </a>
        <a
          alt="twitter icon"
          className="h-6 w-6 mx-2 hover:opacity-75"
          href="https://twitter.com/luacmartins_"
          target="_blank"
        >
          <svg
            className="fill-current h-full w-full"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 97.248c-19.04 8.352-39.328 13.888-60.48 16.576 21.76-12.992 38.368-33.408 46.176-58.016-20.288 12.096-42.688 20.64-66.56 25.408C411.872 60.704 384.416 48 354.464 48c-58.112 0-104.896 47.168-104.896 104.992 0 8.32.704 16.32 2.432 23.936-87.264-4.256-164.48-46.08-216.352-109.792-9.056 15.712-14.368 33.696-14.368 53.056 0 36.352 18.72 68.576 46.624 87.232-16.864-.32-33.408-5.216-47.424-12.928v1.152c0 51.008 36.384 93.376 84.096 103.136-8.544 2.336-17.856 3.456-27.52 3.456-6.72 0-13.504-.384-19.872-1.792 13.6 41.568 52.192 72.128 98.08 73.12-35.712 27.936-81.056 44.768-130.144 44.768-8.608 0-16.864-.384-25.12-1.44C46.496 446.88 101.6 464 161.024 464c193.152 0 298.752-160 298.752-298.688 0-4.64-.16-9.12-.384-13.568 20.832-14.784 38.336-33.248 52.608-54.496z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
  <div
    className="mt-12 sm:px-24 lg:px-0 lg:pt-16 lg:mt-0 lg:w-1/2 lg:ml-104 lg:pb-16"
  >
    <p
      className="font-thin text-lg"
    >
      I'm a passionate software engineer using cutting edge technology to create real-world impact. I am currently looking for remote software engineer opportunities to help companies reach their goals. Previously, I was the CEO of an imports and exports company in Brazil where I restructued operations that led to multimillion dollar profits. I am also passionate about travelling and photography.
    </p>
    <section
      className="mt-16"
    >
      <header
        className="font-semibold text-3xl mt-6 border-b"
      >
        Experience
      </header>
      <div
        className="mt-6"
      >
        <div
          className="mb-6 sm:mb-8"
        >
          <div>
            <span
              className="font-semibold text-gray-700 text-sm"
            >
              CEO
            </span>
            <header
              className="font-semibold text-2xl leading-tight mt-2 sm:mt-0"
            >
              Hadar Importação
            </header>
            <span
              className="text-gray-600 font-hairline text-sm"
            >
              Feb 2012 - Aug 2019 (7y 6mo)
            </span>
            <p
              className="mt-2 font-thin text-lg"
            >
              Hadar is an imports and exports company supplying the textile industry in Brazil. I joined the struggling company in 2011 and during my time as CEO, I was able to restructure all of its operations from product lines and suppliers to logistics and fulfillment partners. I also secured $300k in funding for the growing operations of the company. These changes led the company to an upswing, getting out of the red and reporting over $1.5m in revenue and 28% net profit in the first year. In the following years, I was responsible for bringing 420+ clients and adding 7 new product lines. I left the company in 2019 to pursue my passion in technology.
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      className="mt-16"
    >
      <header
        className="font-semibold text-3xl mt-6 border-b"
      >
        Projects
      </header>
      <div
        className="mt-6"
      >
        <div
          className="mb-6 sm:mb-8"
        >
          <div>
            <div>
              <header
                className="font-semibold text-2xl leading-tight mt-2 sm:mt-0"
              >
                CM Photo
              </header>
              <p
                className="mt-2 font-thin text-lg"
              >
                I designed and wrote the code for my online photography store. I used React's Next.js framework and TailwindCSS to stactically generate the front end. State is managed with React hooks and data with markdown files. Testing is done using Jest. Fulfillment and payment were integrated using Rest APIs.
              </p>
            </div>
            <div
              className="flex justify-start mt-2 py-2 sm:py-3"
            >
              <a
                alt="live demo"
                className="bg-blue-500 rounded px-2 py-1 text-white sm:w-32 sm:text-center hover:shadow hover:bg-blue-600"
                href="website"
                target="_blank"
              >
                Live demo
              </a>
              <a
                alt="source code"
                className="border border-blue-500 px-2 py-1 text-blue-500 rounded ml-4 sm:w-32 sm:text-center hover:shadow hover:text-blue-600 hover:border-blue-600"
                href="github"
                target="_blank"
              >
                Source code
              </a>
            </div>
          </div>
        </div>
        <div
          className="mb-6 sm:mb-8"
        >
          <div>
            <div>
              <header
                className="font-semibold text-2xl leading-tight mt-2 sm:mt-0"
              >
                Ourglobaltrek
              </header>
              <p
                className="mt-2 font-thin text-lg"
              >
                A personal travel blog that details my adventures around the globe. It was originally launched with a custom wordpress theme and I later revamped the old theme. The front end is rendered with React and Sass and data is fetched using the Wordpress Rest API. This was my first React project.
              </p>
            </div>
            <div
              className="flex justify-start mt-2 py-2 sm:py-3"
            >
              <a
                alt="live demo"
                className="bg-blue-500 rounded px-2 py-1 text-white sm:w-32 sm:text-center hover:shadow hover:bg-blue-600"
                href="http://www.ourglobaltrek.com"
                target="_blank"
              >
                Live demo
              </a>
              <a
                alt="source code"
                className="border border-blue-500 px-2 py-1 text-blue-500 rounded ml-4 sm:w-32 sm:text-center hover:shadow hover:text-blue-600 hover:border-blue-600"
                href="https://github.com/luacmartins/wp-theme-ultimate-expedition"
                target="_blank"
              >
                Source code
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      className="mt-16"
    >
      <header
        className="font-semibold text-3xl mt-6 border-b"
      >
        Education
      </header>
      <div
        className="mt-6"
      >
        <div
          className="mb-6 sm:mb-8"
        >
          <div>
            <span
              className="font-semibold text-gray-700 text-sm"
            >
              Full-stack Developer
            </span>
            <header
              className="font-semibold text-2xl leading-tight mt-2 sm:mt-0"
            >
              freeCodeCamp.org
            </header>
            <span
              className="text-gray-600 font-hairline text-sm"
            >
              2020 - Present
            </span>
            <p
              className="mt-2 font-thin text-lg"
            />
          </div>
        </div>
        <div
          className="mb-6 sm:mb-8"
        >
          <div>
            <span
              className="font-semibold text-gray-700 text-sm"
            >
              Bachelor's in Mechanical Engineering
            </span>
            <header
              className="font-semibold text-2xl leading-tight mt-2 sm:mt-0"
            >
              Universidade Federal de Santa Catarina
            </header>
            <span
              className="text-gray-600 font-hairline text-sm"
            >
              2008 - 2013
            </span>
            <p
              className="mt-2 font-thin text-lg"
            />
          </div>
        </div>
        <div
          className="mb-6 sm:mb-8"
        >
          <div>
            <span
              className="font-semibold text-gray-700 text-sm"
            >
              Research Scholar
            </span>
            <header
              className="font-semibold text-2xl leading-tight mt-2 sm:mt-0"
            >
              University of Pennsylvania
            </header>
            <span
              className="text-gray-600 font-hairline text-sm"
            >
              2012 - 2013
            </span>
            <p
              className="mt-2 font-thin text-lg"
            />
          </div>
        </div>
        <div
          className="mb-6 sm:mb-8"
        >
          <div>
            <span
              className="font-semibold text-gray-700 text-sm"
            >
              Exchange Program
            </span>
            <header
              className="font-semibold text-2xl leading-tight mt-2 sm:mt-0"
            >
              Swiss Federal Institute of Technology
            </header>
            <span
              className="text-gray-600 font-hairline text-sm"
            >
              2011 - 2012
            </span>
            <p
              className="mt-2 font-thin text-lg"
            />
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
